subprojects/core/src/main/java/org/gradle/caching/configuration/internal/BuildCacheConfigurationInternal.java

```diff
@@ -41,4 +45,15 @@ public interface BuildCacheConfigurationInternal extends BuildCacheConfiguration
      * Replaces remote build cache.
      */
     void setRemote(@Nullable BuildCache remote);
+
+    /**
+     * Gets build cache service registrations
+     */
+    @Nonnull
```

Comment:
`@Nonnull` is not needed here due to the [package level annotation `@NonNullApi`]( https://github.com/gradle/gradle/blob/06677b922432b742db3b81fb3938616f83b82387/subprojects/core/src/main/java/org/gradle/caching/configuration/internal/package-info.java#L16).

Reply:
oh cool thanks